### PR TITLE
Test/integration test for creating wallet

### PR DESCRIPTION
**Pull request by MackowiakPiotr**

### CODE REVIEW

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        WalletCreateDTO newWallet = new WalletCreateDTO("NewWallet");
+
+        // when
+        ResultActions response = mockMvc.perform(post("/api/wallets")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(newWallet)));
+
+        // then
+        response.andExpect(status().isCreated())
+                .andDo(print())
```
Comment (mirekgab) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java:42]: Please remove the print() method, it should only be used for debugging.

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        WalletCreateDTO newWallet = new WalletCreateDTO("@3H*(G");
+
+        // when
+        ResultActions response = mockMvc.perform(post("/api/wallets")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(newWallet)));
+
+        // then
+        response.andExpect(status().isBadRequest())
+                .andDo(print())
```
Comment (mirekgab) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java:61]: Please remove the print() method, it should only be used for debugging.

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        WalletCreateDTO newWallet = new WalletCreateDTO("nameOfThisWalletIsTooLong");
+
+        // when
+        ResultActions response = mockMvc.perform(post("/api/wallets")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(newWallet)));
+
+        // then
+        response.andExpect(status().isBadRequest())
+                .andDo(print())
```
Comment (mirekgab) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java:85]: Please remove the print() method, it should only be used for debugging.

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                .andExpect(jsonPath("$.status").value("TEA003"))
+                .andExpect(jsonPath("$.message").value("VALIDATION_FAILED"))
+                .andExpect(jsonPath("$.description[0]")
+                        .value("error: field: name, default message: must match \"[\\w ]+\"," +
+                                " rejected value: @3H*(G"))
+                .andExpect(jsonPath("$.statusCode").value(400));
+
```
Comment (mirekgab) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java:68]: When checking the returned value, you can use the value from enum ErrorCode

```java
                .andExpect(jsonPath("$.status").value(ErrorCode.TEA003.getBusinessStatus()))
                .andExpect(jsonPath("$.message").value(ErrorCode.TEA003.getBusinessMessage()))
                .andExpect(jsonPath("$.statusCode").value(ErrorCode.TEA003.getBusinessStatusCode()))
```

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        // then
+        response.andExpect(status().isBadRequest())
+                .andDo(print())
+                .andExpect(jsonPath("$.status").value("TEA003"))
+                .andExpect(jsonPath("$.message").value("VALIDATION_FAILED"))
+                .andExpect(jsonPath("$.description[0]")
+                        .value("error: field: name, default message: size must be between 0 and 20," +
+                                " rejected value: nameOfThisWalletIsTooLong"))
+                .andExpect(jsonPath("$.statusCode").value(400));
+
```
Comment (mirekgab) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java:92]: When checking the returned value, you can use the value from enum ErrorCode

```java
                .andExpect(jsonPath("$.status").value(ErrorCode.TEA003.getBusinessStatus()))
                .andExpect(jsonPath("$.message").value(ErrorCode.TEA003.getBusinessMessage()))
                .andExpect(jsonPath("$.statusCode").value(ErrorCode.TEA003.getBusinessStatusCode()))
```

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java (hunk 2 above)
Comment (MackowiakPiotr) [reply src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java]: done

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java (hunk 3 above)
Comment (MackowiakPiotr) [reply src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java]: done

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java (hunk 4 above)
Comment (MackowiakPiotr) [reply src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java:64]: thank you, done

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java (hunk 5 above)
Comment (MackowiakPiotr) [reply src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java:84]: thank you, done

**Review by mirekgab — DISMISSED** [review]

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+import pl.byczazagroda.trackexpensesappbackend.exception.ErrorCode;
+import pl.byczazagroda.trackexpensesappbackend.repository.WalletRepository;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+public class TestCreateWallet extends BaseIntegrationTestIT {
+
```
Comment (Iwona007) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java:21]: Please delete "public". We use JUnit5, so "public" is defoult for the class and all test methods.

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    @Autowired
+    WalletRepository walletRepository;
+
```
Comment (Iwona007) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java:24]: Please add "private". This is going about encapsulation.

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    @BeforeEach
+    public void clearTestDB() {
+        walletRepository.deleteAll();
```
Comment (Iwona007) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java:27]: The same what in 20 line

**Review by Iwona007 — CHANGES_REQUESTED** [review]

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java (hunk 7 above)
Comment (MackowiakPiotr) [reply src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java:23]: done

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java (hunk 8 above)
Comment (MackowiakPiotr) [reply src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/TestCreateWallet.java:26]: done

**Review by dawciobiel — APPROVED** [review]
Looks good for me

**Review by mirekgab — APPROVED** [review]

**Review by Iwona007 — APPROVED** [review]

**Pull request merged**